### PR TITLE
Move the fluent setup to async [DNMY]

### PR DESCRIPTION
**Pull request by slang25**
This is an effort to get rid of the synchronous AWS APIs (with the exception of `Publish`).

In order to achieve this with fluent api, I've introduce a `.Build()` method that should be awaited, which will perform all of the startup actions asynchronously.

### CODE REVIEW

File: JustSaying.IntegrationTests/AwsTools/SqsQueueIntegrationTests.cs [hunk 1]
```diff
@@ -17,12 +17,12 @@ protected override SqsQueueByName CreateSystemUnderTest()
         {
             QueueUniqueKey = "test" + DateTime.Now.Ticks;
             var queue = new SqsQueueByName(RegionEndpoint.EUWest1, QueueUniqueKey, CreateMeABus.DefaultClientFactory().GetSqsClient(RegionEndpoint.EUWest1), 1, new LoggerFactory());
-            queue.Exists();
+            queue.ExistsAsync().GetAwaiter().GetResult();
```
Comment (AnthonySteele) [inline JustSaying.IntegrationTests/AwsTools/SqsQueueIntegrationTests.cs]: Is it possible to make the setup and teardown async as well?

Comment (slang25) [inline JustSaying.IntegrationTests/AwsTools/SqsQueueIntegrationTests.cs]: I didn't do this cause the base class is in JustBehave, however I could always create a new base class in this project, I'll take a look 👍

**Comment by AnthonySteele** [comment]
It's a lot of code to take in. But I agree with the general ideas though, with deleting the sync wrapper methods of `.Create` etc, and I don't mind the `.Build()` method at the end of the fluent chain.

**Comment by slang25** [comment]
lol, thanks resharper:
```xml
<ItemGroup>
    <Reference Include="JustBehave">
      <HintPath>..\..\..\Users\stuart.lang\.nuget\packages\justbehave\1.0.0.12\lib\net452\JustBehave.dll</HintPath>
    </Reference>
    <Reference Include="Ploeh.AutoFixture">
      <HintPath>..\..\..\Users\stuart.lang\.nuget\packages\autofixture\3.50.2\lib\net40\Ploeh.AutoFixture.dll</HintPath>
    </Reference>
  </ItemGroup>
```

**Comment by slang25** [comment]
@AnthonySteele Done

File: README.md [hunk 2]
```diff
@@ -84,17 +86,17 @@ We currently support SQS subscriptions only, but keep checking back for other me
...
-                // Some logic here ...
-                // e.g. bll.NotifyRestaurantAboutOrder(message.OrderId);
-                return true;
-            }
-        }
-````
+```csharp
+public class OrderNotifier : IHandler<OrderAccepted>
+{
+    public bool Handle(OrderAccepted message)
```
Comment (AnthonySteele) [inline README.md]: I would prefer the first example to use the async version, as the sync `IHandler` is deprecated.

Comment (slang25) [inline README.md]: Good call 👍

Comment (AnthonySteele) [inline README.md]: Also, I know you didn't write this, but " BLL layer" stands for (I think, I'm not sure) "Business Logic Layer" layer?

**Comment by slang25** [comment]
I'm in FPH tomorrow so might drop by to talk about this some more, I'm not particularly happy with the return type from `.Build()`, but I'll explain in full tomorrow.

**Comment by jaimalchohan** [comment]
How are we doing on this? I'm very keen on getting a version 5 out as we're super close to needing to use it in a production setting

**Comment by tonyharverson** [comment]
This is a year old, and not being explored any more.

**Pull request closed without merging**